moveos/moveos-stdlib/moveos-stdlib/sources/account_storage.move

```diff
@@ -115,61 +74,37 @@ module moveos_std::account_storage {
+        borrow_resource_from_account_storage<T>(self)
+    }
 
-    #[private_generics(T)]
-    /// Borrow a resource from the account's storage
-    /// This function equates to `borrow_global<T>(address)` instruction in Move
-    public fun global_borrow<T: key>(ctx: &Context, account: address): &T {
-        let account_storage = borrow_account_storage(ctx, account);
-        borrow_resource_from_account_storage<T>(account_storage)
+    public fun borrow_mut_resource<T: key>(self: &mut AccountStorage): &mut T {
+        borrow_mut_resource_from_account_storage<T>(self)
     }
 
-    #[private_generics(T)]
-    /// Borrow a mut resource from the account's storage
-    /// This function equates to `borrow_global_mut<T>(address)` instruction in Move
-    public fun global_borrow_mut<T: key>(ctx: &mut Context, account: address): &mut T {
-        let account_storage = borrow_account_storage_mut(ctx, account);
-        borrow_mut_resource_from_account_storage<T>(account_storage)
+    public fun move_resource_to<T: key>(self: &mut AccountStorage, resource: T){
+        add_resource_to_account_storage(self, resource);
     }
 
-    #[private_generics(T)]
-    /// Move a resource to the account's storage
-    /// This function equates to `move_to<T>(&signer, resource)` instruction in Move
-    public fun global_move_to<T: key>(ctx: &mut Context, account: &signer, resource: T){
-        let account_address = signer::address_of(account);
-        //Auto create the account storage when move resource to the account
-        ensure_account_storage(ctx, account_address);
-        let account_storage = borrow_account_storage_mut(ctx, account_address);
-        add_resource_to_account_storage(account_storage, resource);
+    public fun move_resource_from<T: key>(self: &mut AccountStorage): T {
+        remove_resource_from_account_storage(self)
     }
 
-    #[private_generics(T)]
-    /// Move a resource from the account's storage
-    /// This function equates to `move_from<T>(address)` instruction in Move
-    public fun global_move_from<T: key>(ctx: &mut Context, account: address): T {
-        let account_storage = borrow_account_storage_mut(ctx, account);
-        remove_resource_from_account_storage<T>(account_storage)
+    public fun exists_resource<T: key>(self: &AccountStorage) : bool {
+        exists_resource_at_account_storage<T>(self)
     }
 
-    #[private_generics(T)]
-    /// Check if the account has a resource of the given type
-    /// This function equates to `exists<T>(address)` instruction in Move
-    public fun global_exists<T: key>(ctx: &Context, account: address) : bool {
-        if (exist_account_storage(ctx, account)) {
-            let account_storage = borrow_account_storage(ctx, account);
-            exists_resource_at_account_storage<T>(account_storage)
-        }else{
-            false
-        }
+    public(friend) fun transfer(obj: Object<AccountStorage>, account: address) {
```

Comment:
This should be `transfer_extend`?

Reply:
AccountStorage does not support public transfer and does not allow to extend.